### PR TITLE
NAS-115593 / None / WIP NAS-115593: Add support for dynamic forms in ix-forms, part 3

### PR DESCRIPTION
**Pull request by bvasilenko**
I just completed cleaning up the code.
In next PR, I will commit more unit-tests.

### CODE REVIEW

File: src/app/interfaces/dynamic-form-schema.interface.ts [hunk 1]
```diff
@@ -57,7 +63,7 @@ export interface DynamicFormSchemaDict extends DynamicFormSchemaBase {
 
 export interface AddListItemEmitter {
```
Comment (undsoft) [inline src/app/interfaces/dynamic-form-schema.interface.ts:64]: `AddListItemEvent`

File: src/app/interfaces/dynamic-form-schema.interface.ts [hunk 2]
```diff
@@ -23,31 +23,37 @@ export interface DynamicFormSchemaBase {
...
 export interface DynamicFormSchemaCheckbox extends DynamicFormSchemaBase {
   type: 'checkbox';
+  tooltip?: string;
 }
 
 export interface DynamicFormSchemaList extends DynamicFormSchemaBase {
   type: 'list';
   items?: DynamicFormSchemaNode[];
-  items_schema?: any[];
+  items_schema?: unknown[];
```
Comment (undsoft) [inline src/app/interfaces/dynamic-form-schema.interface.ts:56]: itemsSchema

File: src/app/pages/applications/forms/chart-form/chart-form.component.ts [hunk 3]
```diff
@@ -25,35 +30,36 @@ import { IxSlideInService } from 'app/services/ix-slide-in.service';
 export class ChartFormComponent {
   title: string;
   config: any;
```
Comment (undsoft) [inline src/app/pages/applications/forms/chart-form/chart-form.component.ts:32]: Can we type this?

File: src/app/pages/applications/forms/chart-form/chart-form.component.ts [hunk 4]
```diff
@@ -25,35 +30,36 @@ import { IxSlideInService } from 'app/services/ix-slide-in.service';
...
+
   isLoading = false;
+  isNew = true;
   dynamicSection: DynamicFormSchema[] = [];
   dialogRef: MatDialogRef<EntityJobComponent>;
 
-  form = this.formBuilder.group({
-    release_name: ['', Validators.required],
-  });
+  form = this.formBuilder.group({});
```
Comment (undsoft) [inline src/app/pages/applications/forms/chart-form/chart-form.component.ts:41]: At the very least can be typed to something like:

```
interface ChartFormValues {
  release_name: string;
  version?: string;
  [key: string]: string | number | boolean;
}
```

File: src/app/pages/applications/forms/chart-form/chart-form.component.ts [hunk 5]
```diff
@@ -86,9 +92,73 @@ export class ChartFormComponent {
     }
   }
 
+  setChartCreate(chart: CatalogApp): void {
```
Comment (undsoft) [inline src/app/pages/applications/forms/chart-form/chart-form.component.ts:95]: A bit similar to `setChartEdit`. Maybe extract some logic into smaller methods?

**Review by undsoft — COMMENTED** [review]
* Items in lists need more padding to fix their layout.
* When editing a chart in old code, it seems to be sending much less fields than the new code. I'm not necessarily saying this is correct, just something to check. Maybe default values are not supposed to be saved?

Other than that looks good.

**Comment by bvasilenko** [comment]
@undsoft 
I've addressed all the review remarks in the other draft PR https://github.com/truenas/webui/pull/6685

**Pull request closed without merging**